src/renderer/components/+cluster/cluster-metric-switchers.tsx

```diff
@@ -24,32 +24,50 @@ const NonInjectedClusterMetricSwitchers = observer(({
   nodeStore,
 }: Dependencies) => {
   const { masterNodes, workerNodes } = nodeStore;
-  const metricsValues = clusterOverviewStore.getMetricsValues(clusterOverviewStore.metrics);
-  const disableRoles = !masterNodes.length || !workerNodes.length;
-  const disableMetrics = !metricsValues.length;
+  const { cpuUsage, memoryUsage } = clusterOverviewStore.metrics;
+  const hasMasterNodes = masterNodes.length > 0;
+  const hasWorkerNodes = workerNodes.length > 0;
+  const hasCpuMetrics = normalizeMetrics(cpuUsage).data.result[0].values.length > 0;
+  const hasMemoryMetrics = normalizeMetrics(memoryUsage).data.result[0].values.length > 0;
 
   return (
     <div className="flex gaps" style={{ marginBottom: "calc(var(--margin) * 2)" }}>
       <div className="box grow">
         <RadioGroup
           asButtons
-          className={cssNames("RadioGroup flex gaps", { disabled: disableRoles })}
+          className="RadioGroup flex gaps"
           value={clusterOverviewStore.metricNodeRole}
           onChange={metric => clusterOverviewStore.metricNodeRole = metric}
         >
-          <Radio label="Master" value={MetricNodeRole.MASTER}/>
-          <Radio label="Worker" value={MetricNodeRole.WORKER}/>
+          <Radio
+            label="Master"
+            value={MetricNodeRole.MASTER}
+            disabled={!hasMasterNodes}
+          />
+          <Radio
+            label="Worker"
+            value={MetricNodeRole.WORKER}
+            disabled={!hasWorkerNodes}
+          />
         </RadioGroup>
       </div>
       <div className="box grow" style={{ textAlign: "right" }}>
         <RadioGroup
           asButtons
-          className={cssNames("RadioGroup flex gaps", { disabled: disableMetrics })}
+          className="RadioGroup flex gaps"
```

Comment:
We could probably getting rid from such global classnames like `flex gaps` and moving toward css-modules, where we:

- we replace those classes with pure css-rules, e.g. classname old global `<div className="Item flex box grow"` becomes css-rule `.Item { display: flex; flex: 1; }`
- re-use some corresponding classes from `tailwindcss`, e.g. `.Item { @apply flex flex-grow; }`

But this is out of scope for current PR maybe and should be considered as suggestion for all future PRs.